.github/workflows/publish-docker.yaml

```diff
@@ -35,9 +35,19 @@ jobs:
     env:
       VERSION: ${{ github.sha }}
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
         with:
           submodules: true
+      - name: Setup Golang
+        uses: actions/setup-go@v4
+        with:
+          go-version: 1.18
+      - name: Pull deps for tools/go-agent
```

Comment:
Small trick here, we pull deps for tools/go-agent which internally uses the dep cache and then package it into vendor module which allows to be reused in the docker build.